opensrp-giz-malawi/src/main/java/org/smartregister/giz/task/SaveReasonForDefaultingEventTask.java

```diff
@@ -0,0 +1,131 @@
+import org.smartregister.clientandeventmodel.Event;
+import org.smartregister.clientandeventmodel.FormEntityConstants;
+import org.smartregister.domain.Client;
+import org.smartregister.domain.db.EventClient;
+import org.smartregister.giz.application.GizMalawiApplication;
+import org.smartregister.repository.EventClientRepository;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Iterator;
+
+import timber.log.Timber;
+
+public class SaveReasonForDefaultingEventTask extends AsyncTask<Void, Void, Void> {
+    private final String jsonString;
+    private final String locationId;
+    private final String baseEntityId;
+    private final String providerId;
+    private final EventClientRepository eventClientRepository;
+
+    public SaveReasonForDefaultingEventTask(String jsonString, String locationId, String baseEntityId, String providerId,
+                                            EventClientRepository eventClientRepository) {
+        this.jsonString = jsonString;
+        this.locationId = locationId;
+        this.baseEntityId = baseEntityId;
+        this.providerId = providerId;
+        this.eventClientRepository = eventClientRepository;
+    }
+
+    @Override
+    protected Void doInBackground(Void... params) {
+        try {
+            processReasonForDefaultingEvent();
+        } catch (Exception e) {
+            Timber.e(Log.getStackTraceString(e));
+        }
+
+        return null;
+    }
+
+    private void processReasonForDefaultingEvent() throws Exception {
+        JSONObject jsonForm = new JSONObject(jsonString);
+
+
+        JSONArray fields = ChildJsonFormUtils.fields(jsonForm);
+        if (fields == null) {
+            return;
+        }
+
+        String encounterDateField = ChildJsonFormUtils.getFieldValue(fields, Constants.DATE_REACTION);
+        String encounterType = ChildJsonFormUtils.getString(jsonForm, ChildJsonFormUtils.ENCOUNTER_TYPE);
+        JSONObject metadata = ChildJsonFormUtils.getJSONObject(jsonForm, ChildJsonFormUtils.METADATA);
+
+        Date encounterDate = new Date();
+        if (StringUtils.isNotBlank(encounterDateField)) {
+            Date dateTime = ChildJsonFormUtils.formatDate(encounterDateField, false);
+            if (dateTime != null) {
+                encounterDate = dateTime;
+            }
+        }
+
+        Event event = (Event) new Event()
+                .withBaseEntityId(baseEntityId) //should be different for main and subform
+                .withEventDate(encounterDate)
+                .withEventType(encounterType)
+                .withLocationId(locationId)
+                .withProviderId(providerId).withEntityType(Constants.CHILD_TYPE)
+                .withChildLocationId(ChildLibrary.getInstance().context().allSharedPreferences().fetchCurrentLocality())
+                .withFormSubmissionId(ChildJsonFormUtils.generateRandomUUIDString()).withDateCreated(new Date());
+
+
+        //add metadata
+        addMetadata(fields, event, metadata);
```

Comment:
Why not use `JsonFormUtils.createEvent` from core.

Reply:
Replaced